audb/core/utils.py

```diff
@@ -36,6 +37,46 @@ def lookup_backend(
     return _lookup(name, version)[1]
 
 
+def md5_read_chunk(
+        fp: typing.IO,
+        chunk_size: int = 8192,
+):
+    while True:
+        data = fp.read(chunk_size)
+        if not data:
+            break
+        yield data
+
+
+def md5(
+        path: str,
+        chunk_size: int = 8192,
+) -> str:
+    r"""Create MD5 checksum of file or folder."""
+    path = audeer.path(path)
+
+    if not os.path.isdir(path):
+        return audbackend.md5(path)
+
+    files = audeer.list_file_names(
+        audeer.path(path),
+        recursive=True,
+        hidden=True,
+        basenames=True,
+    )
+
+    hasher = hashlib.md5()
+    for file in files:
+        # encode file name that renaming of files
+        # produces different checksum
+        hasher.update(file.encode())
```

Comment:
Is this safe against file names that contain umlauts on Windows vs. Linux?

Reply:
I don't know :)

Reply:
I added a test in `tests/test_utils.py`, which should fail with the current implementation.

Reply:
I now pushed a fix where `os.path.sep` is replaced with `/`. The test also covers umlaute. So if it passes, we should hopefully be safe.